src/compiler/types.ts

```diff
@@ -2893,6 +2901,61 @@ namespace ts {
         failedLookupLocations: string[];
     }
 
+    export type LintErrorMethod = {
+        (err: string): void;
+        (err: string, span: Node): void;
+        (err: string, start: number, length: number): void;
+    };
+    export type LintStopMethod = () => void;
+
+    /*
+    * Walkers call stop to halt recursion into the node's children
+    * Walkers call error to add errors to the output.
+    */
+    export interface LintWalker {
+        visit(node: Node, stop: LintStopMethod, error: LintErrorMethod): void;
+    }
+
+    export interface SyntacticLintProviderStatic {
+        new (typescript: typeof ts, args: any): LintWalker;
+    }
+
+    export interface SemanticLintProviderStatic {
+        new (typescript: typeof ts, checker: TypeChecker, args: any): LintWalker;
+    }
+
+    export namespace ExtensionKind {

```

Comment:
Do we really need this pseudo-string-valued-enum? Can we just use the string literals themselves to avoid having to look up this value on `ExtensionKind` at runtime when a string literal would be more efficient?